src/compiler/checker/lines_4000_8000.rs

```diff
@@ -292,13 +292,12 @@ impl NodeBuilder {
             return factory
                 .create_literal_type_node(
                     &self.synthetic_factory,
-                    &*Into::<Rc<Node>>::into(
-                        if type_.as_intrinsic_type().intrinsic_name() == "true" {
-                            factory.create_true(&self.synthetic_factory)
-                        } else {
-                            factory.create_false(&self.synthetic_factory)
-                        },
-                    ),
+                    if type_.as_intrinsic_type().intrinsic_name() == "true" {
```

Comment:
I'd previously commented that here was a place where I had to make sure to wrap the node in an `Rc<Node>` before passing along, I think it makes more sense for the `.create_literal_type_node()` method to just take an `Rc<Node>` since that's what it's storing and then there's no way to forget to do that